app/controllers/users_controller.rb

```diff
@@ -1,6 +1,49 @@
 class UsersController < ApplicationController
+
+  before_action :authenticate_user!
+
+  # Only users and admins can edit user's profile
+  def edit
+  end
+
+  def show
+     @user = User.find(params[:id])
+  end
+
+  # Only users and admins can update user's profile
+  def update
+    if current_user.update_attributes(user_params)
+      redirect_to edit_user_registration_path
+  end
+
+  private
+    # Using a private method to encapsulate the permissable parameters
+
+    # Finds the User with the ID stored in the session with the key 
+    # :current_user_id. Logging in sets the session value and logging
+    # out removes it.
+    def current_user
+      @current_user || session[:current_user_id] &&
+        User.find_by(id: session[:current_user_id])
+      end
```

Comment:
This looks like it has an extra end.   Also, `current_user` should be populated by devise, we shouldn't need to build it.